extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/AbstractJob.java

```diff
@@ -14,7 +14,12 @@
  */
 public abstract class AbstractJob implements Job {
 
-    protected String thingUID;
+    private final String thingUID;
+
+    public AbstractJob(String thingUID) {
+        checkArgument(thingUID != null, "Thing UID cannot be null");
```

Comment:
instead of `cannot be null` I would prefer `must not be null` in error messages which point the caller of this method to his failure. An instance returned by a method `cannot be null` if it will never be null. A method parameter can be null any time but `must not be null` to guarantee the correct behaviour of the method.